src/Helper/Product/Utility.php

```diff
@@ -198,19 +204,26 @@ public function setVisibility($product)
     /**
      * @param MageProduct $product
      *
-     * @return \Magento\CatalogInventory\Api\Data\StockItemInterface
+     * @return array
      */
     public function getStockItem($product)
     {
-        $stockItem = $this->stockItemFactory->create();
+        $stockId       = $this->getStockIdForCurrentWebsite->execute();
+        $stockItemData = $this->getStockItemData->execute($product->getSku(), $stockId);
```

Comment:
the execute throws exceptions, catch or log it if it is possible that it might get thrown or just annotate in docblock.

Reply:
Pls. check